homeassistant/components/dsmr/sensor.py

```diff
@@ -143,8 +143,7 @@ async def connect_and_reconnect():
                 transport, protocol = await hass.loop.create_task(reader_factory())
             except (
                 serial.serialutil.SerialException,
-                ConnectionRefusedError,
-                TimeoutError,
+                OSError,
```

Comment:
I would keep the ConnectionRefusedError and TimeoutError, it seems both can be thrown by the supporting library so just adding OSError would suffice.

Reply:
I'll add them back for documentation purpose then. Both are subclasses of OSError so functionality wise it is not really needed https://docs.python.org/3/library/exceptions.html#exception-hierarchy

Reply:
If OSError covers them it's fine. Note that I added in another PR (#37464) the asyncio.TimeoutError, which seems to be different from TimeoutError.

Reply:
Yeah I noticed that one to. I did not include it in this PR to avoid duplicate changes